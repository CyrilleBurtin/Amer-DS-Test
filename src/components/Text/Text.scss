.slm-text {
  @apply text-salomon-black
    font-semibold
    text-base

    lg:text-xl;
}
