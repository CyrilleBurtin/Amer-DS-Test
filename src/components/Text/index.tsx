import { ReactNode } from 'react'
import { classNames } from '../../utils'
import './Text.scss'

interface Text {
  className?: string
  children: string | ReactNode
}

const Text = ({ className, children }: Text) => {
  const classes = {
    text: classNames('slm-text', className)
  }

  return <p className={classes.text}>{children}</p>
}

export default Text
