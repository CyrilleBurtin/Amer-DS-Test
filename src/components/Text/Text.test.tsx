import Text from './index.js';
import { render } from '@testing-library/react';

describe('<Text>', () => {
  it('should render the default component with custom text', () => {
    const { container, getByText } = render(<Text>Custom text</Text>);

    expect(container.firstChild).toHaveClass('slm-text');
    expect(getByText(/Custom text/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render the default component with custom text and custom class', () => {
    const { container, getByText } = render(
      <Text className="p-4">Custom text</Text>
    );

    expect(container.firstChild).toHaveClass('slm-text p-4');
    expect(getByText(/Custom text/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
