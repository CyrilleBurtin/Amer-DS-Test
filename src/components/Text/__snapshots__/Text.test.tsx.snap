// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Text> > should render the default component with custom text 1`] = `
<div>
  <p
    class="slm-text"
  >
    Custom text
  </p>
</div>
`;

exports[`<Text> > should render the default component with custom text and custom class 1`] = `
<div>
  <p
    class="slm-text p-4"
  >
    Custom text
  </p>
</div>
`;
